### ISSUE TITLE
entrenar CUDA training forward: NaN/Inf on Blackwell, CPU fallback on Ada Lovelace

### ISSUE THREAD

**Issue by noahgift** [paiml/aprender#563]
## Summary

`apr finetune` training forward pass fails on both tested GPU architectures:

1. **yoga (sm_89, Ada Lovelace)**: CUDA forward fails silently, every step falls back to CPU → 36 tok/s (186x slower than unsloth 6,697 tok/s)
2. **gx10 (sm_121, Blackwell)**: CUDA forward succeeds but produces NaN/Inf loss → backward pass skipped → loss stuck at 11.93 (no learning)

### yoga (sm_89)
```
[CUDA] GPU forward failed, falling back to CPU for this step
```
Repeats every step. Training completes on CPU at 0.075 samples/sec.

### gx10 (sm_121)
```
[CUDA] NaN/Inf loss detected — skipping backward pass
```
Q_proj GEMM succeeds (TRACE shows correct output), but something downstream produces NaN/Inf.

## Root Cause Analysis

The training forward path in `entrenar/src/finetune/instruct_pipeline.rs:728` calls `forward_logits_gpu_resident()`. This chains through:
1. `forward_cuda_training()` — transformer block forward
2. `rms_norm_forward()` — final layer norm
3. `gemm_forward()` — lm_head logits

On sm_89: Step 1 fails (likely PTX kernel execution after cuLinkCreate failure — see paiml/trueno#231).
On sm_121: Steps 1-3 succeed numerically but the resulting logits contain NaN/Inf (possible fp16 overflow in lm_head GEMM with vocab_size=151936).

## Impact

- apr finetune cannot produce a trained model on any tested hardware
- 186x throughput deficit vs unsloth (Python QLoRA) on same GPU
- Blocks Sovereign Stack training parity goal

## Contract

Provable contract: `qwen-train-canary/contracts/cuda-training-forward-v1.yaml`
Falsification tests: FALSIFY-TRAIN-001, FALSIFY-TRAIN-002, FALSIFY-TRAIN-003

## Refs

- PMAT-420 (qwen-train-canary)
- paiml/trueno#231 (PTX loading)
- F-WL-06 in training canary spec
